linuxkit/linuxkit
# Upgrade runc to 1.0.3

**- What I did**

- Update runc package run 1.0.3
- Remove the RO workaround that prevented container start
- update all examples and yamls to use images of new runc (with help from @djs55)

**- How I did it**

- update the version number
- revert the workaround patch (git revert)
- run scripts/update-component-sha.sh

**- How to verify it**
First ensure the RO issues is still a reality on runc 1.0.2 by checking out only the revert (only the first patch of this PR) and then using the command below :
 `cd ~/linuxkit/src/cmd/linuxkit && go build . && ./linuxkit build ../../../linuxkit.yml && ./linuxkit run qemu -kernel linuxkit`

The system will boot and show error.
```
INFO[2021-12-21T15:26:42.136250559Z] shim disconnected                             id=rngd
ERRO[2021-12-21T15:26:42.136718017Z] copy shim log                                 error="read /proc/self/fd/23: file already closed"
ERRO[0000] failed to create task                         error="OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting \"devpts\" to rootfs at \"/dev/pts\" caused: mkdir /containers/services/rngd/rootfs/dev/pts: read-only file system: unknown"
time="2021-12-21T15:26:42.141640559Z" level=info msg="starting signal loop" namespace=services.linuxkit path=/run/containerd/io.containerd.runtime.v2.task/services.linuxkit/shutdown pid=525
```
We can see in `ctr -n services.linuxkit t ls` that rngd service failed to start.

Then checking out this PR to upgrade runc and upgrade the runc package SHA in all recipes, and run the command line again.
No error message appears.
The rngd container started (and stopped) properly.

**- Description for the changelog**

Upgrade runc to 1.0.3

**- A picture of a cute animal (not mandatory but encouraged)**
